docs/testplan/HA-SmartSwitch-test-plan.md

```diff
@@ -0,0 +1,188 @@
+| Active T1-T0 link drop – Active         | Verify packet flow when T1-T0 link drop.                      | • Start  I/O through active side.<br>• Configure DPU1 side T1-T0 link drop.                       | DPU1 becomes non-active, DPU2 becomes standalone. | T2 receives packets with 1 allowed disruption.  |
+| Active T1-T0 link drop – Standby        | Verify packet flow when T1-T0 link drop.                      | • Start  I/O through standby side.<br>• Configure DPU1 side T1-T0 link drop.                      | DPU1 becomes non-active, DPU2 becomes standalone. | T2 receives packets with 1 allowed disruption.  |
+| Standby T1-T0 link drop – Active        | Verify packet flow when T1-T0 link drop.                      | • Start  I/O through active side.<br>• Configure DPU2 side T1-T0 link drop.                       | DPU1 becomes standalone, DPU2 is anything but active. | T2 receives packets without disruption.          |
+| Standby T1-T0 link drop - Standby      | Verify packet flow when T1-T0 link drop.                      | • Start  I/O through standby side.<br>• Configure DPU2 side T1-T0 link drop.                      | DPU1 becomes standalone, DPU2 is anything but active. | T2 receives packets without disruption.          |
+
+### Module 6 Critical Process Crash 
+For all process crash cases, we will have 4 variations, it’s 
+1. Process crash on DPU1, traffic landing on DPU2
+2. Process crash on DPU2, traffic landing on DPU1
+3. Process crash on DPU1, traffic landing on DPU1
+4. Process crash on DPU2, traffic landing on DPU2
+The expected behavior is same, that HA state remains unchanged. No data plane disruption is expected. 
+
+| Case            | Goal                                     | Test Steps                                            | Expected Control Plane Behavior                   | Expected Data Plane Behavior                      |
+|-----------------|------------------------------------------|-------------------------------------------------------|--------------------------------------------------|---------------------------------------------------|
+| syncd on DPU    | Verify when syncd crash on DPU.         | • Start  I/O<br>• Kill syncd on DPU       | DPU1 remains active, DPU2 remains standby.        | T2 receives packets without disruption.          |
+| hamgrd on NPU   | Verify when hamgrd crash on NPU.        | • Start  I/O<br>• Kill hamgrd on NPU      | DPU1 remains active, DPU2 remains standby.        | T2 receives packets without disruption.          |
+| pmon on NPU     | Verify when pmon crash on NPU.          | • Start  I/O<br>• Kill pmon on NPU        | DPU1 remains active, DPU2 remains standby.        | T2 receives packets without disruption.          |
+| bgpd on NPU     | Verify when bgpd crash on NPU.          | • Start  I/O<br>• Kill bgpd on NPU        | DPU1 remains active, DPU2 remains standby.        | T2 receives packets without disruption.          |
```

Comment:
this section is all unplanned events, it is really hard to achieve no packet drops. the current goal for unplanned events are "time from detection to mitigation" is 2s.

Reply:
Addressed.